sklearn/metrics/tests/test_score_objects.py

```diff
@@ -47,7 +51,7 @@ class EstimatorWithoutFit(object):
     pass
 
 
-class EstimatorWithFit(object):
+class EstimatorWithFit(BaseEstimator):
```

Comment:
Why is this required? Would it fail if `EstimatorWithFit` would not inherit from the `BaseEstimator` class? I would like to make sure that we are not enforcing the inheritance on `EstimatorWithFit` for 3rd party libraries.